### DISCUSSION TITLE
Lazy load large nested parts of a document when expanded

### DISCUSSION THREAD

**Discussion by adi-fi-ma** [josdejong/svelte-jsoneditor#501] [Ideas]
Hi, 

I was wondering whether there's an option to retrieve the expanded path once clicking on that little expand arrow button?
I would like to use this ability in order to lazy-load large json on-demand to my website and need to get the expanded path by an event (on click). Is there any way to access the path via the component props? Is there any event that is triggered in that case which I can use?
Thanks!

**Comment by josdejong** [comment]
That is an interesting idea. Some first thoughts:

It may be possible to use a [JavaScript Proxy](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy) for this. I'm not sure how this will work out though and whether it will work at all. There may be some changes needed inside the JSON editor itself. Also, in `text` mode you'll have to load the full document anyway, so switching to text mode will probably be slow. It may be worth an experiment. Anyone interested in trying out what happens?

An other direction could be to implement an API that gives you control over the expanded nodes, so you get a trigger when a node is expanded and you can then load the nested data lazily. I'm a bit in doubt on whether that can work out nicely, since the loading will be async but there will be no way to let the user know that something is loading, and also, I don't think that will work in `text` mode.

Another idea is to implement lazy loading as a first-class citizen. So, you could show a button "load nested content", which will asynchronously load the data when clicked (showing a spinner whilst loading). This may be relatively complex to implement, I'm not sure.

**Reply by josdejong** [reply]
I've done a little experiment with `Proxy`, but this doesn't play nice with the editor. The reason is that the editor is built upon immutable data structures, wheras `Proxy` requires the opposite: making edits in the original data.
